tests/test_benchmark.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+from os import path, listdir
+from conftest import skipif
+from devito.exceptions import CompilationError
+from devito import configuration
+from subprocess import CalledProcessError, check_call
+
+
+pytestmark = skipif(['yask', 'ops'])
+
+
+@pytest.mark.parametrize('mode', ['bench'])
+@pytest.mark.parametrize('problem', ['acoustic', 'tti', 'elastic', 'viscoelastic'])
+def test_bench(mode, problem):
+    """
+    Test combinations of modes x problems and created .json filename.
+    Testing for both OpenMP enabled or disabled.
+    """
+    if configuration['openmp']:
+        nthreads = configuration['platform'].cores_physical
+    else:
+        nthreads = 1
+
+    command_bench = ['python', 'benchmarks/user/benchmark.py', mode, '-P', problem,
+                     '-d', '40', '40', '40', '--tn', '20', '-x', '1']
+    try:
+        check_call(command_bench)
+    except CalledProcessError as e:
+        raise CompilationError('Command "%s" return error status %d. '
+                               'Unable to compile code.\n' %
+                               (e.cmd, e.returncode))
+    dir_name = 'results/'
+
+    base_filename = problem
```

Comment:
**all** the underscores in the list of variables below should be dropped

then you just do:

```
bench_file = path.join('_'.join(....))
```

Reply:
this is probably wrong? It creates this: `a_c_o_u_s_t_i_c_a_r_c_h_[_u_n_k_n_o_w_n_]_s_`

Reply:
ok, found the way